File: API_Favoritos/Models/DTO/CatSubCatModel.cs
namespace API_Favoritos.Models.DTO
{
    public class CatSubCatModel
    {
        public int CatCodi { get; set; }
        public string? CatDesc { get; set; }
        public int? CatPai { get; set; } 
        public List<CatSubCatModel> SubCategorias { get; set; } = new List<CatSubCatModel>();
        public List<SiteModel> Sites { get; set; } = new List<SiteModel>(); 
    }
}

File: favoritos/src/app/models/CatSubCat.Model.ts
import { SiteModel } from './Site.Model';

export class CatSubCatModel {
  CatCodi: number;
  CatDesc: string;
  CatPai?: number;
  SubCategorias: CatSubCatModel[];
  Sites: SiteModel[];

  constructor() {
    this.CatCodi = 0;
    this.CatDesc = '';
    this.CatPai = undefined;
    this.SubCategorias = [];
    this.Sites = [];
  }
}
